sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/random.scala

```diff
@@ -85,7 +85,7 @@ case class Randn(seed: Long) extends RDG {
     val rngTerm = ctx.freshName("rng")
     val className = classOf[XORShiftRandom].getName
     ctx.addMutableState(className, rngTerm,
-      s"$rngTerm = new $className($seed + org.apache.spark.TaskContext.getPartitionId());")
+      s"$rngTerm = new $className(${seed}L + org.apache.spark.TaskContext.getPartitionId());")
```

Comment:
this is an existing and small bug, sometimes `seed` is large and can not be represented as int literal, add a `L` at end to make it a long literal.